src/Features/Core/Diagnostics/BaseDiagnosticIncrementalAnalyzer.cs

```diff
@@ -19,24 +19,145 @@ protected BaseDiagnosticIncrementalAnalyzer(Workspace workspace)
         }
 
         #region IIncrementalAnalyzer
+        /// <summary>
+        /// Analyze a single document such that local diagnostics for that document become available,
+        /// prioritizing analyzing this document over analyzing the rest of the project.
+        /// Calls <see cref="DiagnosticAnalyzerService.RaiseDiagnosticsUpdated(object, DiagnosticsUpdatedArgs)"/> for each
+        /// unique group of diagnostics, where a group is identified by analysis classification (syntax/semantics), document, and analyzer.
+        /// </summary>
+        /// <param name="document"></param>
```

Comment:
Can you remove the <param> nodes if there is nothing interesting to say about them?